### PR TITLE
Fix multiple IN interpolation

### LINKED ISSUE #152: Support multiple IN interpolation in queries
Follow-up issue of https://github.com/zio/zio-jdbc/issues/104 solved in v0.1.0.

I tried to use the interpolation in a query that contains two `IN` clauses and fails whereas having a single `IN` clause works.

Example of query:
```scala
val field1List: List[String] = List("a", "b")
val field2List: List[Int] = List(2, 3)
val startDate: String = ???
val endDate: String = ???

sql"""SELECT group_id, count(*) as my_count
        FROM my_table
        WHERE date >= $startDate AND date < $endDate
        AND field1 IN ($field1List)
        AND field2 IN ($field2List)
        GROUP BY group_id"""
  .query[(Long, Long)]
  .selectAll
```

The error I get is:
```
Parameter "#6" is not set; SQL statement:
  SELECT group_id, count(*) as my_count
                  FROM my_table
                  WHERE date >= ? AND date < ?
                  AND field1 IN (?, ?)
                  AND field2 IN (?, ?)
                  GROUP BY group_id [90012-220]
  		at org.h2.message.DbException.getJdbcSQLException(DbException.java:658)
```

If I remove one of the `IN` clauses, it does work.

### PR DESCRIPTION
**Pull request by lvitaly**
Closes #152

### CODE REVIEW

File: core/src/test/scala/zio/jdbc/SqlFragmentSpec.scala [hunk 1]
```diff
@@ -159,6 +159,13 @@ object SqlFragmentSpec extends ZIOSpecDefault {
                 assertIn(List(1, 2, 3)) &&
                 assertIn(Vector(1, 2, 3)) &&
                 assertIn(Set(1, 2, 3))
+              } + test("multiple interpolation params") {
```
Comment (gaeljw) [inline core/src/test/scala/zio/jdbc/SqlFragmentSpec.scala:162]: Maybe worth testing with more than 2? Didn't read the implementation but 2 items can often be managed in a way that 3+ items wouldn't work.

File: core/src/main/scala/zio/jdbc/SqlFragment.scala [hunk 2]
```diff
@@ -113,27 +113,28 @@ sealed trait SqlFragment { self =>
...
-          )
 
         case _ =>
-          sql.append("?")
           paramsBuilder += param.value.toString
+          size += 1
       }
+      sql.append(
+        if (size == 1) "?"
+        else Seq.fill(size)("?").mkString(",")
```
Comment (gaeljw) [inline core/src/main/scala/zio/jdbc/SqlFragment.scala:136]: The changes in this file are not related to the PR, are they? Not sure they're bringing much value IMHO.

File: core/src/main/scala/zio/jdbc/SqlFragment.scala [hunk 3]
```diff
@@ -113,27 +113,28 @@ sealed trait SqlFragment { self =>
...
-            Seq.fill(array.length)("?").mkString(",")
-          )
 
         case _ =>
-          sql.append("?")
           paramsBuilder += param.value.toString
+          size += 1
       }
+      sql.append(
+        if (size == 1) "?"
```
Comment (gaeljw) [inline core/src/main/scala/zio/jdbc/SqlFragment.scala:135]: The if is not needed, is it? `mkstring` won't apply if size is 1 anyway.

**Review by gaeljw — APPROVED** [review]

File: core/src/main/scala/zio/jdbc/SqlFragment.scala (hunk 3 above)
Comment (lvitaly) [reply core/src/main/scala/zio/jdbc/SqlFragment.scala:135]: You are right. `mkString` will handle a single-item case, but it will require additional allocations, which we can easily drop with the `if` condition. I assume that a multi-item placeholder is much more rare than a single-item placeholder. So, I think this "micro" optimization is excused.

File: core/src/test/scala/zio/jdbc/SqlFragmentSpec.scala (hunk 1 above)
Comment (lvitaly) [reply core/src/test/scala/zio/jdbc/SqlFragmentSpec.scala:162]: No. It's not necessary. The root cause of the issue was in the parameter indexing of the `ZConnection.executeSqlWith` function. Now, it covers sequences correctly, and quantity doesn't matter.

Comment (gaeljw) [reply core/src/test/scala/zio/jdbc/SqlFragmentSpec.scala:162]: My point is that it's not necessary with the current implementation but a future change could make it necessary to detect a bug. Up to you but I prefer to always use 3+ items in such tests.

Comment (lvitaly) [reply core/src/test/scala/zio/jdbc/SqlFragmentSpec.scala]: I added more items to the existing test and added one more to cover the empty collection.

**Review by adamgfraser — APPROVED** [review]
